pkg/controller/snapshotter/snapshotter.go

```diff
@@ -774,8 +918,9 @@ func (vs *volumeSnapshotter) UpdateVolumeSnapshotData(snapshotDataName string, s
 		return fmt.Errorf("Error: expecting type VolumeSnapshotData but received type %T", objCopy)
 	}
 
-	snapshotDataCopy.Status.Conditions = *status
-	glog.Infof("Updating VolumeSnapshotData object")
+	ind := len(*status) - 1
+	snapshotDataCopy.Status.Conditions = append(snapshotDataCopy.Status.Conditions, (*status)[ind])
```

Comment:
Same as with VolumeSnapshot: I think the status conditions array should contain exactly one condition in our case.